p2p/src/conn.rs

```diff
@@ -261,13 +261,14 @@ impl Tracker {
 /// Start listening on the provided connection and wraps it. Does not hang
 /// the current thread, instead just returns a future and the Connection
 /// itself.
-pub fn listen<H>(
-	stream: Stream,
+pub fn listen<S, H>(
+	stream: S,
 	version: ProtocolVersion,
 	tracker: Arc<Tracker>,
 	handler: H,
 ) -> io::Result<(ConnHandle, StopHandle)>
 where
+	S: Stream + 'static,
```

Comment:
For my own edification, why is the `'static` lifetime needed here and other places? Does that affect the lifetime of the concrete object?